### PR TITLE
Bump tusd to v2.4.0

### PR DESCRIPTION
**Pull request by kobergj**
Bumps tusd to v2.4.0 to enable tackling https://github.com/owncloud/ocis/issues/1755

### CODE REVIEW

**Comment by kobergj** [comment]
Full-ci ocis pipeline is green: https://github.com/owncloud/ocis/pull/9714

**Comment by 2403905** [comment]
If I understand correctly we also have to implement the  Unwrap interface otherwise we getting warnings regarding NetworkControlError/NetworkTimeoutError and "feature not supported"

https://tus.github.io/tusd/advanced-topics/usage-package/#i-am-getting-warnings-regarding-networkcontrolerrornetworktimeouterror-and-feature-not-supported-why

File: pkg/rhttp/datatx/manager/tus/tus.go [hunk 1]
```diff
@@ -99,7 +98,8 @@ func (m *manager) Handler(fs storage.FS) (http.Handler, error) {
 	config := tusd.Config{
 		StoreComposer:         composer,
 		NotifyCompleteUploads: true,
-		Logger:                log.New(appctx.GetLogger(context.Background()), "", 0),
+		// TODO: bring back!
```
Comment (dragonchaser) [inline pkg/rhttp/datatx/manager/tus/tus.go:101]: Is that on purpose?

**Review by dragonchaser — DISMISSED** [review]

File: pkg/rhttp/datatx/manager/tus/tus.go (hunk 1 above)
Comment (kobergj) [reply pkg/rhttp/datatx/manager/tus/tus.go:101]: Good Question. The problem is that the logger interface changed in v2. It now uses https://pkg.go.dev/golang.org/x/exp/slog Afaikt there a two option how to solve it

A) We adjust our logger to follow new interface. I think we'd need to implement https://pkg.go.dev/golang.org/x/exp/slog#Handler

B) We use their default logger.

Is it worth going for A?

Comment (kobergj) [reply pkg/rhttp/datatx/manager/tus/tus.go]: Brought back the logger - but it is still noop.

**Comment by kobergj** [comment]
> If I understand correctly we also have to implement the Unwrap interface otherwise we getting warnings regarding NetworkControlError/NetworkTimeoutError and "feature not supported"

Correct, but we are passing a noop logger (with did that before too). Hence we will not see those error messages.

**Review by fschade — APPROVED** [review]

**Pull request merged**